### PR TITLE
Adapt payments module

### PR DESCRIPTION
**Pull request by OmeGak**

### CODE REVIEW

File: indico/modules/events/registration/controllers/display.py [hunk 1]
```diff
@@ -155,6 +155,8 @@ def _process(self):
             user_data.update((attr, getattr(self.invitation, attr)) for attr in ('first_name', 'last_name', 'email'))
         return self.view_class.render_template('display/regform_display.html', self.event, event=self.event,
                                                sections=get_event_section_data(self.regform), regform=self.regform,
+                                               payment_conditions=event_settings.get(self.event, 'conditions'),
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/display.py]: The payment settings should be imported like this:

    from indico.modules.payment import event_settings as payment_event_settings

Right now it's absolutely not clear which settings you have without looking at the imports (eventually we should rename it in payment/__init__.py and other modules defining settings proxies like this). Not sure what I was thinking when not using proper names while adding those...

File: indico/modules/payment/controllers.py [hunk 2]
```diff
@@ -188,24 +185,16 @@ class RHPaymentEventCheckout(RHPaymentEventBase):
...
-        if not self._registrant.isCheckoutSessionAlive():
-            self._registrant._checkout_attempt_dt = datetime.now()
-        else:
-            checkout_attempt_dt = self._registrant.getCheckoutAttemptDt()
-            checkout_attempt_delta = int(math.ceil((datetime.now() - checkout_attempt_dt).total_seconds() / 60))
-        event = self._conf
-        amount = self._registrant.getTotal()
-        currency = event_settings.get(event, 'currency')
-        plugins = get_active_payment_plugins(event)
+        if self.registration.state.name != 'unpaid':
```
Comment (ThiefMaster) [inline indico/modules/payment/controllers.py]: why not use the state enum here? name checks are meant for templates where it's easier that way..

File: indico/modules/payment/models/transactions.py [hunk 3]
```diff
@@ -196,55 +190,37 @@ class PaymentTransaction(db.Model):
...
         return self.provider == '_manual'
 
     @return_ascii
     def __repr__(self):
         # in case of a new object we might not have the default status set
         status = TransactionStatus(self.status).name if self.status is not None else None
-        return '<PaymentTransaction({}, {}, {}, {}, {}, {} {}, {})>'.format(self.id, self.event_id, self.registrant_id,
-                                                                            status, self.provider, self.amount,
-                                                                            self.currency, self.timestamp)
+        return '<PaymentTransaction({}, {}, {} {}, {}, {})>'.format(status, self.provider, self.amount, self.currency,
```
Comment (ThiefMaster) [inline indico/modules/payment/models/transactions.py]: looks like a good `format_repr` candidate

File: indico/modules/payment/templates/event_checkout.html [hunk 4]
```diff
@@ -52,10 +47,9 @@ <h2 class="page-title">{% trans %}Checkout{% endtrans %}</h2>
         // Simply call reloadPaymentMethod({your: 'custom data'}) in the payment form template of your plugin and
         // override render_payment_form in the plugin class.
         global.reloadPaymentMethod = function reloadPaymentMethod(extraData) {
-            var data = {{ registrant_params | tojson }};
-            data.method = paymentMethod;
+            var data = {method: paymentMethod} ;
```
Comment (ThiefMaster) [inline indico/modules/payment/templates/event_checkout.html]: suddenly a wild space appeared

File: migrations/versions/201510221159_1d741fc6586c_backup_old_transaction_table.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+Create Date: 2015-10-22 11:59:18.116917
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from indico.core.db.sqlalchemy import PyIntEnum
+from indico.core.db.sqlalchemy import UTCDateTime
+from indico.core.db.sqlalchemy.util.bulk_rename import _rename_constraint
+from indico.modules.payment.models.transactions import TransactionStatus
+from sqlalchemy.dialects import postgresql
```
Comment (ThiefMaster) [inline migrations/versions/201510221159_1d741fc6586c_backup_old_transaction_table.py]: before indico imports, then a linebreak

File: migrations/versions/201510221323_2f4eefa1050c_delete_old_transaction_table.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+Revises: 1d741fc6586c
+Create Date: 2015-10-22 13:23:42.602410
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from indico.core.db.sqlalchemy import PyIntEnum
+from indico.core.db.sqlalchemy import UTCDateTime
+from indico.modules.payment.models.transactions import TransactionStatus
+from sqlalchemy.dialects import postgresql
```
Comment (ThiefMaster) [inline migrations/versions/201510221323_2f4eefa1050c_delete_old_transaction_table.py]: same here

File: indico/modules/payment/controllers.py (hunk 2 above)
Comment (OmeGak) [inline indico/modules/payment/controllers.py]: :+1:

File: indico/modules/events/registration/templates/management/regform_list.html [hunk 7]
```diff
@@ -12,6 +12,29 @@
...
+                <div class="icon icon-coins"></div>
+                <div class="text">
+                    <div class="label">
+                        {% trans %}Payments disabled{% endtrans %}
+                    </div>
+                    {% trans %}Payments are disabled for this event{% endtrans %}
+                </div>
+                <div class="toolbar right">
+                    <a href="#" class="js-enable-payments"
+                       data-href="{{ url_for('event_features.switch', event, feature='payment') }}"
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/management/regform_list.html]: The beauty of having RESTful endpoints :)

Comment (OmeGak) [inline indico/modules/events/registration/templates/management/regform_list.html]: Although switch is the element in features, toggle is the action. Who should I blame for lousy naming? :P

Comment (ThiefMaster) [inline indico/modules/events/registration/templates/management/regform_list.html]: That comment wasn't criticism ;)

Comment (OmeGak) [inline indico/modules/events/registration/templates/management/regform_list.html]: I know. :point_up:

**Pull request merged**